pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -87,23 +185,55 @@ final class Event {
       : eventName = DashEvent.hotReloadTime,
         eventData = {'timeMs': timeMs};
 
-  Event.lintUsageCounts({required String usageCounts})
-      : eventName = DashEvent.lintUsageCounts,
+  /// Event that is emitted periodically to report the number of times each lint
+  /// has been enabled.
+  ///
+  /// [count] - the number of contexts in which the lint was enabled
+  ///
+  /// [name] - the name of the lint
+  Event.lintUsageCount({
+    required int count,
+    required String name,
+  })  : eventName = DashEvent.lintUsageCount,
         eventData = {
-          'usageCounts': usageCounts,
+          'count': count,
+          'name': name,
         };
 
+  /// Event that is emitted periodically to report the amount of memory being
+  /// used.
+  ///
+  /// [rss] - the resident set size in megabytes
+  ///
+  /// If this is not the first time memory has been reported for this isession,
+  /// then the following parameters should be included:
+  ///
+  /// [periodSec] - the number of seconds since the last memory usage data was
+  ///     gathered
+  ///
+  /// [mbPerSec] - the number of megabytes of memory that were added or
+  ///     subtracted per second since the last report
   Event.memoryInfo({
-    String? rss,
+    required int rss,
     int? periodSec,
     double? mbPerSec,
   })  : eventName = DashEvent.memoryInfo,
         eventData = {
-          if (rss != null) 'rss': rss,
+          'rss': rss,
           if (periodSec != null) 'periodSec': periodSec,
           if (mbPerSec != null) 'mbPerSec': mbPerSec
         };
```

Comment:
@polina-c can you confirm that the event you added is covered with these parameters? I believe we may have included a `type` parameter as well that had `delta` and `initial` as possible values? Linked the PR below

- https://github.com/dart-lang/tools/pull/92#discussion_r1192815640

Reply:
I don't know whether the server code is currently correct, but these are the only parameters being set in the code that's sending the event (https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/lib/src/analytics/analytics_manager.dart#L201).

Reply:
Thank you for highlighting where that event was used. Seems like we didn't implement the `type` parameter. I suggest that we should include the parameter so that querying the difference between those two fields is fairly trivial on the backend.